src/main/java/jupyter/AsDisplayData.java

```diff
@@ -0,0 +1,19 @@
+package jupyter;
+
+import java.util.Map;
+
+/**
+ * Convenience interface for classes whose display data is known in advance.
+ *
+ * Classes implementing this interface are displayed via the {@link AsDisplayData#display} method
+ * by default, without needing to register a {@link Displayer} in advance.
+ */
+public interface AsDisplayData {
+    Map<String, String> display();
+
+    Displayer<AsDisplayData> displayer = new Displayer<AsDisplayData>() {
```

Comment:
I don't see a reason for this to be public. Why not move it to a top-level class and restrict it to package-private for Registration?

Reply:
Yes, moved that to `Registration` (and made it private).